src/logging.cpp

```diff
@@ -273,17 +318,45 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
     if (m_print_to_file) {
         assert(m_fileout != nullptr);
 
-        // reopen the log file, if requested
-        if (m_reopen_file) {
-            m_reopen_file = false;
+        auto reopen_file = [&]() {
             FILE* new_fileout = fsbridge::fopen(m_file_path, "a");
             if (new_fileout) {
                 setbuf(new_fileout, nullptr); // unbuffered
                 fclose(m_fileout);
                 m_fileout = new_fileout;
+                m_file_size = 0;
             }
+        };
+
+        // reopen the log file, if requested
+        if (m_reopen_file) {
+            m_reopen_file = false;
+            reopen_file();
```

Comment:
If `logrotate` is being used, this will happen (sending a `SIGHUP` to cause `debug.log` to be reopened), but in that case the user should disable this PR's log rotation (by setting `m_rotate_keep` to zero in the configuration). But if they forget (definitely need release notes on this), I _think_ it will be okay, because either this PR's log rotation will always happen (since it waits only until `debug.log` reaches 1 MB by default), or the Unix `logrotate` will happen (if the file size threshold is less than 1 MB). It would be confusing if both were happening! Maybe more thought is needed here.